examples/cosmos/ics20/ics20.qnt

```diff
@@ -90,20 +90,22 @@ module ics20 {
   /// state and the acknowledgment
   pure def onRecvPacket(chainState: ChainState, packet: Packet): (ChainState, FungibleTokenPacketAcknowledgement) = {
     pure val data = packet.data
-    pure val result = if (movingBackAlongTrace({port: packet.sourcePort, channel: packet.sourceChannel}, data.denom)) {
+    pure val movingBack = movingBackAlongTrace(data.denom, { port: packet.sourcePort, channel: packet.sourceChannel })
+    pure val newDenom = updateTrace(movingBack, data.denom, { port: packet.destPort, channel: packet.destChannel })
+
+    pure val result = if (movingBack) {
```

Comment:
Can we add a type annotation for `result`. I found this hard to read. Perhaps also add the suggested comment?
```suggestion
    // result is a pair of 1. a BankResult and 2. an error message that is used in case BankResult.success = false 
    pure val result = if (movingBack) {
```

Reply:
I agree, pairs are hard to read in general.
I think we should either use records over pairs, or immediately unpack the elements of a returned pair into their own named bindings.